app/code/Magento/Review/Ui/DataProvider/Product/ReviewDataProvider.php

```diff
@@ -79,10 +81,25 @@ public function getData()
     }
 
     /**
-     * {@inheritdoc}
+     * Apply sorting if it set
+     *
+     * @return void
+     */
+    private function applySorting(): void
+    {
+        $sorting = $this->request->getParam('sorting');
+        if (is_array($sorting)) {
+            $select = $this->getCollection()->getSelect();
+            $select->order($sorting['field'] . ' ' . $sorting['direction']);
```

Comment:
```suggestion
            $this->getCollection()->setOrder($sorting['field'], $sorting['direction']);
```
Apparently, there are some edge cases, it seems better to use the collection provided method than adding it directly to the select.